src/compressor/Compressor.cc

```diff
@@ -59,6 +59,21 @@ boost::optional<Compressor::CompressionMode> Compressor::get_comp_mode_type(cons
 
 CompressorRef Compressor::create(CephContext *cct, const std::string &type)
 {
+  // support "random" for teuthology testing
+  if (type == "random") {
+    static std::random_device seed;
+    static std::default_random_engine engine(seed());
+    static Spinlock mutex;
+
+    int alg = COMP_ALG_NONE;
+    std::uniform_int_distribution<> dist(0, COMP_ALG_LAST - 1);
+    {
+      std::lock_guard<Spinlock> lock(mutex);
+      alg = dist(engine);
+    }
```

Comment:
seeing a lot of spam like this: `/usr/lib64/ceph/libceph_none.so: cannot open shared object file: No such file or directory`

if we get `alg == COMP_ALG_NONE` here, it should probably return nullptr directly